File: MPP/proiect/java/model/src/main/java/services/IService.java
package services;

import model.Purchase;
import model.dtos.ShowData;
import utils.IObservable;

import java.util.Collection;

/**
 * Created by archie on 3/28/2017.
 */
public interface IService extends IObservable<Integer> {
    void addPurchase(Purchase entity);

    Collection<ShowData> getAll();

    boolean login(String username, String password);
}

File: MPP/proiect/c#/Festival/Service/IService.cs
using System.Collections.Generic;
using Festival.Model;

namespace Festival.Service
{
    public interface IService
    {
        event changeOccuredType ChangeOccured; //tine locul clientului deoarece are loc un singur event

        void AddPurchase(int showId, string clientName, int quantity);
        bool Login(string username, string password);
        IEnumerable<ShowData> GetAll();
    }
}